FpsCheck.py

```diff
@@ -366,33 +364,48 @@ def find_invalid_well_known(self, check_sets):
-                            field_sym_difference = [json_schema["primary"], 
-                            curr_fps_set.primary]
-                        else:
-                            field_sym_difference = []
-                    else:
-                        field_sym_difference = set(json_schema[field] 
-                                if field in json_schema.keys() else []) ^ set(
-                                    curr_fps_set.relevant_fields_dict[field])
-                        if field == 'ccTLDs':
-                            for aliased_site in json_schema[field]:
-                                field_sym_difference.update(
-                                    set(json_schema[field][aliased_site]) ^ 
-                                    set(
-                                    curr_fps_set.relevant_fields_dict[field]
-                                    [aliased_site]))
-                    if field_sym_difference:
-                        self.error_list.append("The following member(s) of " 
-                        + field + " were not present in both the changelist "
-                        + "and .well-known/first-party-set.json file: " + 
-                        str(sorted(field_sym_difference)))
+                well_known_set = FpsSet(
+                    json_schema.get('ccTLDs'), 
+                    json_schema.get('primary'), 
+                    json_schema.get('associatedSites'), 
+                    json_schema.get('serviceSites'))
+                if well_known_set.primary != curr_fps_set.primary:
+                    field_sym_difference = [well_known_set.primary, curr_fps_set.primary]
+                    self.error_list.append("The following primary was not "
+                    + "present in both the changelist and "
+                    + ".well-known/first-party-set.json file: "
+                    + str(sorted(field_sym_difference)))
+                if well_known_set.associated_sites != curr_fps_set.associated_sites:
+                    field_sym_difference = set(
+                       well_known_set.associated_sites) ^ set(curr_fps_set.associated_sites)
+                    self.error_list.append("The following member(s) of " 
+                    + "associatedSites were not present in both the "
+                    + "changelist and .well-known/first-party-set.json "
+                    + "file: " + str(sorted(field_sym_difference)))
+                if well_known_set.service_sites != curr_fps_set.service_sites:
+                    field_sym_difference = set(
+                       well_known_set.service_sites) ^ set(curr_fps_set.service_sites)
+                    self.error_list.append("The following member(s) of " 
+                    + "serviceSites were not present in both the "
+                    + "changelist and .well-known/first-party-set.json "
+                    + "file: " + str(sorted(field_sym_difference)))
```

Comment:
Looks like this code is repeated for the associated sites, service sites, and then for each list of ccTLD aliases. Consider factoring out a helper function (or method on the FpsSet class) to construct the error messages. E.g. something like:

```python
def compute_error_from_lists(subset_name, list1, list2):
    if list1 == list2:
        return []
    diff = set(list1) ^ set(list2)   # could be improved if we choose.
    return [f"The following member(s) of {subset_name} were not present in both the PR and the {WELL_KNOWN} file: {str(sorted(diff))}"]
```

Then use it like:

```python
self.error_list.extend(compute_error_from_lists("associatedSites", well_known_set.associated_sites, curr_fps_set.associated_sites))
self.error_list.extend(compute_error_from_lists("serviceSites", well_known_set.service_sites, curr_fps_set.service_sites))
# etc. for ccTLD aliases
```